### ISSUE TITLE
Incorrect output for errors in beforeEach, when nested in a describe group and other groups exist

### ISSUE THREAD

**Issue by fran-f** [pestphp/pest#869]
First of all, thank you for the great 2.9 release! 🙇 I immediately started using `describe` groups, and stumbled on an odd interaction. I'm using Pest 2.9.3 and PHP 8.2.8.

Usually, if a `beforeEach` callback causes an error, related tests fail and the errors are logged:

```php
<?php
describe('first describe', function() {
  beforeEach(fn() => thisWillCauseAnError());
  it('test in the first describe', function() {
    expect(1)->toBe(1);
  });
});
```

```
   FAIL  Tests\AsExpectedTest
  ⨯ first describe → it test in the first describe
  ────────────────────────────────────────────────────────────────────────────────────────────────
   FAILED  Tests\AsExpectedTest > `first describe` → it test in the first describe          Error
  Call to undefined function thisWillCauseAnError()
```

However, if there are other `describe` blocks in the file, the output uses the wrong `describe` and test descriptions

```php
<?php
describe('first group should fail', function() {
  beforeEach(fn() => thisWillCauseAnError());
  it('test in the first describe', function() {
    expect(1)->toBe(1);
  });
});

describe('second group work fine', function() {
  it('test in the second describe', function() {
    expect(1)->toBe(1);
  });
});
```
Pest will report the second group as both failing and passing:
```
   FAIL  Tests\IssueTest
  ⨯ second group work fine → it test in the second describe
  ✓ second group work fine → it test in the second describe
  ────────────────────────────────────────────────────────────────────────────────────────────────
   FAILED  Tests\IssueTest > `second group work fine` → it test in the second describe      Error
  Call to undefined function thisWillCauseAnError()

  at tests\IssueTest.php:3
      1▕ <?php
      2▕ describe('first group should fail', function() {
  ➜   3▕   beforeEach(fn() => thisWillCauseAnError());
      4▕   it('test in the first describe', function() {
```

Pest produces the same output if I swap the order of the two: the failing test will use the descriptions of the preceding `describe` and test.

The correct output in the case above would be:
```
   FAIL  Tests\IssueTest
  ⨯ first group should fail → it test in the first describe
  ✓ second group work fine → it test in the second describe
```

**Comment by nunomaduro** [comment]
Fixed on `v2.9.4`.

**Comment by fran-f** [comment]
That was fast 😮 — Thank you!
